### PR TITLE
Turn layout generation into an option + drop SimpleLayout

### PR DESCRIPTION
**Pull request by gwwatkin**
Usage:
```
--layoutgenerator, -L  Automatically generates a layout for the given number of qubits. Incompatible with -l. Options:
                            - compact (default): Uses Litinski's Game of Surace Code compact layout (https://arxiv.org/abs/1808.02892)
                            - edpc: Uses a layout specified in the EDPC paper by Beverland et. al. (https://arxiv.org/abs/2110.11493)
```

### TODO
 - [x] Update the NPM api. Waiting for #46 to go through

### CODE REVIEW

File: npm/src/index.d.ts [hunk 1]
```diff
@@ -1,21 +1,23 @@
-type inputType = 'lli' | 'qasm';
```
Comment (gwwatkin) [inline npm/src/index.d.ts:1]: Looks like there were `\r` line endings that were auto removed,have to hide white space to view

**Review by alexnguyenn — CHANGES_REQUESTED** [review]
Need to update docstring for `npm/src/index.ts` as well

File: npm/src/index.js [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
+ * Runs the slicer from string input.
+ * 
+ * @param {string} input Circuit content as string
+ * @param {string} inputType 'qasm' for QASM input, 'lli' (default) for LLI input.
+ * @param {string} layoutGenerator Choose the type of layout generator: 'compact' (default), 'edpc'
+ * @param {string} cnotCorrections Add Xs and Zs to correct the the negative outcomes: 'never' (default), 'always'
+ * @returns {{err: string, exit_code: number, output: string}} Slicer result
+ * 
+ */
+async function runSlicer(input, inputType='lli', layoutGenerator=null, cnotCorrections='never') {
```
Comment (alexnguyenn) [inline npm/src/index.js:13]:
```suggestion
async function runSlicer(input, inputType='lli', layoutGenerator='compact', cnotCorrections='never') {
```
Docstring mentions default is 'compact'. If you are planning on having `null` then I think `''` is better

Comment (gwwatkin) [reply npm/src/index.js:13]: Was trying to delegate the choice of the default to the executable and not have to replicate the logic here

Comment (gwwatkin) [reply npm/src/index.js:13]: I'm ok with `''` was keeping null for consistency with the `?` in the type signature

**Review by alexnguyenn — APPROVED** [review]

**Pull request merged**